tasks/upgrades/utils.ts

```diff
@@ -0,0 +1,58 @@
+import { HardhatRuntimeEnvironment } from 'hardhat/types'
+import { IComponents } from '../../common/configuration'
+import { IDeployments } from '../../scripts/deployment/common'
+import { isValidContract } from '../../common/blockchain-utils'
+
+export const validateDeployments = async (
+  hre: HardhatRuntimeEnvironment,
+  deployments: IDeployments,
+  version: string
+) => {
+  // Check trading lib defined
+  if (!deployments.tradingLib) {
+    throw new Error(
+      `Missing deployed TradingLib for version ${version} in network ${hre.network.name}`
+    )
+  } else if (!(await isValidContract(hre, deployments.tradingLib))) {
+    throw new Error(`TradingLib contract not found in network ${hre.network.name}`)
+  }
+
+  // Check Main implementation is defined
+  if (!deployments.implementations.main) {
+    throw new Error(
+      `Missing deployed Main implementation for version ${version} in network ${hre.network.name}`
+    )
+  } else if (!(await isValidContract(hre, deployments.implementations.main))) {
+    throw new Error(`TradingLib contract not found in network ${hre.network.name}`)
+  }
+
+  // Check all componet implementations are defined
+  const componentsKeys: string[] = [
+    'backingManager',
+    'basketHandler',
+    'broker',
+    'distributor',
+    'furnace',
+    'rTokenTrader',
+    'rsrTrader',
+    'rToken',
+    'stRSR',
+  ]
+
+  componentsKeys.forEach(async (keystr) => {
+    if (!deployments.implementations.components[keystr as keyof IComponents]) {
+      throw new Error(
+        `Missing deployed ${keystr} implementation for version ${version} in network ${hre.network.name}`
+      )
+    } else if (
+      !(await isValidContract(
+        hre,
+        deployments.implementations.components[keystr as keyof IComponents]
+      ))
+    ) {
```

Comment:
else

Reply:
Else is the expected behavior, just move forward